### PR TITLE
Error update

### PR DESCRIPTION
**Pull request by samspeer**

### CODE REVIEW

**Comment by timothycobrien** [comment]
Going to wait for other PR to get merged in first @samspeer, will revisit in morning

**Comment by timothycobrien** [comment]
Do you have time to rebase @samspeer ?

File: client/src/components/game.tsx [hunk 1]
```diff
@@ -143,12 +143,30 @@ const QuestionBox = ({
...
       start: false,
     });
+    updateUserStats();
     setEndGame(true);
     // game should end for all players
     socket.emit("end", gameInfo.id.toString());
     socket.offAny();
   };
 
+  const updateUserStats = async () => {
```
Comment (timothycobrien) [inline client/src/components/game.tsx:153]: This trigger should almost certainly be server-side

File: client/src/components/profile.tsx [hunk 2]
```diff
@@ -153,52 +152,41 @@ const StatsBlock = ({userInfo, mode}) => {
...
+            method: 'GET',
+            body: mode.toString(),
+            credentials: 'include',
+            headers: {
+                'content-type': 'application/json'
+            }
+        });
+        if(res.ok) {
+            const data = await res.json();
+            return ([
```
Comment (timothycobrien) [inline client/src/components/profile.tsx:165]: Needs rounding

File: client/src/components/profile.tsx [hunk 3]
```diff
@@ -153,52 +152,41 @@ const StatsBlock = ({userInfo, mode}) => {
...
 
-    //grab all the user stats from given mode!
     const [stats, updateStats] = useState(
         [
-            ["Average Speed", 20],
-            ["Level", 30],
-            ["Ranking", 20],
-            ["Win Rate", 20],
-            ["Accuracy", 20],
+            ["Number of Games", Math.random()*100],
```
Comment (timothycobrien) [inline client/src/components/profile.tsx:184]: Also needs rounding

File: client/src/components/profile.tsx (hunk 2 above)
Comment (timothycobrien) [reply client/src/components/profile.tsx:165]: Should maybe do the calculations client side? Have server send raw numbers, might be more efficient

File: server/src/api/v1/game.py [hunk 4]
```diff
@@ -95,6 +95,54 @@ def create_game():
...
+    player: GamePlayer
+    for player in player_query:
+        statsTable_query = StatsTable.query.filter_by(player_id=player.player_id, mode=game.mode)
+        try:
+            stats = statsTable_query.one()
+        except NoResultFound:
+            stats = StatsTable(player_id=player.player_id, mode=game.mode)
+            db.session.add(stats)
+            db.session.commit()
+        win: boolean = (player.score == max_score)
```
Comment (timothycobrien) [inline server/src/api/v1/game.py:116]: This should be converted to an int

File: server/src/db/models/user.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 from typing import Dict
 from flask_login import UserMixin
+from sqlalchemy.orm import relationship
```
Comment (timothycobrien) [inline server/src/db/models/user.py:3]: Oh sorry this is a holdover from me

**Review by timothycobrien — COMMENTED** [review]
@EvanW1999 PTAL

File: client/src/components/game.tsx (hunk 1 above)
Comment (EvanW1999) [reply client/src/components/game.tsx:153]: Right, so the server side already knows when the game is over (when the "end" event fires). It is also given the game_id of the game that just ended. That's the point when the server should calculate all of the stats for every player that was in that game. You shouldn't have every client manually telling the server to do this.

File: client/src/components/profile.tsx [hunk 6]
```diff
@@ -153,52 +152,41 @@ const StatsBlock = ({userInfo, mode}) => {
...
-        //     console.log("Did not work!");
-        //     return ([
-        //         ["Average Speed", Math.random()*100],
-        //         ["Level", Math.random()*100],
-        //         ["Ranking", Math.random()*100],
-        //         ["Win Rate", Math.random()*100],
-        //         ["Accuracy", Math.random()*100]
-        //     ]);
-        // }
+        const res = await fetch('/api/v1/game/getStats', {
```
Comment (EvanW1999) [inline client/src/components/profile.tsx:155]: Why's this the API? If this all goes on the profile page, might as well just update /api/v1/profile to give all of this data or to add another API "/api/v1/profile/stats"

Comment (samspeer) [reply client/src/components/profile.tsx]: I moved to session.py

File: server/src/db/models/user.py (hunk 5 above)
Comment (samspeer) [reply server/src/db/models/user.py]: I deleted

File: server/src/api/v1/game.py (hunk 4 above)
Comment (samspeer) [reply server/src/api/v1/game.py]: done

File: client/src/components/profile.tsx (hunk 3 above)
Comment (samspeer) [reply client/src/components/profile.tsx]: got rid of randoms and rounded all calculated stats

File: client/src/components/profile.tsx (hunk 2 above)
Comment (samspeer) [reply client/src/components/profile.tsx:165]: I did rounding client side

File: client/src/components/game.tsx (hunk 1 above)
Comment (samspeer) [reply client/src/components/game.tsx]: I made it server side trigger

**Review by EvanW1999 — APPROVED** [review]

**Pull request merged**